var/spack/repos/builtin/packages/hpx/package.py

```diff
@@ -166,4 +166,8 @@ def cmake_args(self):
                 self.define('OTF2_ROOT', spec['otf2'].prefix),
             ]
 
+            # it seems like there was a bug in the default version of APEX in 1.5.x
+            if spec.satisfies("@:1.5"):
```

Comment:
Am I right in thinking this is satisfied by any version 1.5.0 or above? Including say 1.7.0? For 1.6.0 APEX 2.3.0 happens to be correct as well, but I think it'd be better to restrict this to only 1.5.X. Apologies if this in fact does only specify 1.5.X (spack version specifiers are not quite second nature to me yet).

Reply:
Sorry, you pointed out the right thing. I think I should remove the colon and leave just `@1.5`, but I'm going to check documentation exactly how to deal with that. I'll update soon!

Reply:
`@:1.5` is the open range `[0, 1.6.0)`, so 1.6.0 excluded; `@1.5` is a bit ambiguous, since `1.5` can be a concrete version or it could mean a range

Reply:
```
Spack version 0.16.0
Python 3.7.8, Darwin x86_64
>>> from spack.version import ver
>>> [test_ver.satisfies(ver(":1.5")) for test_ver in map(ver, ["1.4", "1.5", "1.5.1", "1.6"])]
[True, True, True, False]
>>> [test_ver.satisfies(ver("1.5")) for test_ver in map(ver, ["1.4", "1.5", "1.5.1", "1.6"])]
[False, True, True, False]
```

So, if I get it correctly, we want the latter behavior, right?

Reply:
Ah, yes, "@1.5" looks like what we want here (there are no versions in HPX without the minor version number, so "1.5" would never be a concrete version).

PS. I read it first as "@1.5:" and not "@:1.5" which probably added even more confusion!

Reply:
Did this now go in with `:1.5`? Did I still misunderstand? :P We do *not* want this applied to 1.4.X (or lower for that matter).